plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/apsadmin/portal/PageSettingsActionAspect.java

```diff
@@ -154,65 +151,58 @@ protected String readFile(String path, PageSettingsAction action) {
-            result = joinPoint.proceed();
-            //se il salvataggio va a buon fine, aggiorna l'oggetto
-            if (null != result && result instanceof String) {
-                result = joinPoint.proceed();
-                String robotContent = request.getParameter(PARAM_ROBOT_CONTENT_CODE);
-                String alternativePath = request.getParameter(PARAM_ROBOT_ALTERNATIVE_PATH_CODE);
-                InputStream is = (!StringUtils.isBlank(robotContent))
-                        ? new ByteArrayInputStream(robotContent.getBytes("UTF-8")) : null;
-                if (StringUtils.isBlank(alternativePath)) {
-                    //default PATH
-                    if (null != is) {
-                        this.getStorageManager().saveFile(ROBOT_FILENAME, false, is);
-                    } else {
-                        this.getStorageManager().deleteFile(ROBOT_FILENAME, false);
-                    }
-                } else if (!PageSettingsUtils.isRightPath(alternativePath)) {
-                    action.addFieldError(PARAM_ROBOT_ALTERNATIVE_PATH_CODE,
-                            action.getText("jpseo.error.robotFilePath.invalid", new String[]{alternativePath}));
-                } else if (this.checkPath(alternativePath, action)) {
-                    if (null != is) {
-                        //alternativePath and string
-                        this.saveFile(alternativePath, is, action);
-                    } else {
-                        File file = new File(alternativePath);
-                        try {
-                            if (file.exists()) {
-                                file.delete();
-                            }
-                        } catch (Exception e) {
-                            logger.error("error deleting file {}", alternativePath);
-                        }
-                    }
+            String robotContent = request.getParameter(PARAM_ROBOT_CONTENT_CODE);
+            String alternativePath = request.getParameter(PARAM_ROBOT_ALTERNATIVE_PATH_CODE);
+            InputStream is = (!StringUtils.isBlank(robotContent))
+                    ? new ByteArrayInputStream(robotContent.getBytes("UTF-8")) : null;
+            if (StringUtils.isBlank(alternativePath)) {
+                //default PATH
+                if (null != is) {
+                    this.getStorageManager().saveFile(ROBOT_FILENAME, false, is);
+                } else {
+                    this.getStorageManager().deleteFile(ROBOT_FILENAME, false);
                 }
-                String xmlParams = this.getConfigManager().getConfigItem(SystemConstants.CONFIG_ITEM_PARAMS);
-                Map<String, String> systemParams = SystemParamsUtils.getParams(xmlParams);
-                if (!StringUtils.isEmpty(alternativePath)) {
-                    systemParams.put(JpseoSystemConstants.ROBOT_ALTERNATIVE_PATH_PARAM_NAME, alternativePath);
+            } else if (!PageSettingsUtils.isRightPath(alternativePath)) {
```

Comment:
Is there a test for this already? Seems like a lot of logic

Reply:
Done